### PR TITLE
Add complex demo scenarios.

### PR DESCRIPTION
**Pull request by felidadae**
NOTE READY TO BE MERGED: contains owca submodule commit not yet merged.

Add two complex demo scenarios:
* last level cache (LLC) contention demo,
* Memory bandwidth (MB) contention demo.

Additional changes:
* ..
* ..

### CODE REVIEW

File: prm/demo_scenarios/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Demo scenarios
+....................
+This document describes how to run demo scenarios on a mesos/aurora cluster.
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
This document describes how to run demo scenarios on a Apache Mesos/Aurora cluster.
```

File: prm/demo_scenarios/README.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Assumptions:
+
+- working directory: PRM project top level directory,
+- copied and filled the template file `demo_scenarios/common/common.template.yaml` to `demo_scenarios/common/common.yaml`,
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]: `demo_scenarios/common/common.template.yaml` should link to the file.

File: prm/demo_scenarios/README.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- the nodes contains owca.pex file with PRM module included.
+
+
+
+1. export needed shell variables:
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
1. Export required environment variables:
```

File: prm/demo_scenarios/README.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+.. code-block:: sh
+
+    export REMOTE_USER=<user>
+    export REMOTE_IP=<remote_ip>
+    export cinventory=demo_scenarios/common/common.template.yaml     # inventory with list of hosts; please copy template file and fill hosts
+    export playbook=owca/workloads/run_workloads.yaml
+
+
+
+2. kill all jobs, stop agant and clean a temporary model data:
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
2. Kill all jobs, stop agent and remove a temporary model data:
```

File: prm/demo_scenarios/README.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+.. code-block:: sh
+
+    ansible-playbook -l $REMOTE_IP -i $REMOTE_IP, $playbook --tags=clean_jobs -v
+    ansible -u $REMOTE_USER -b all -i $REMOTE_IP, -msystemd -a'name=owca state=stopped'
+    ansible -u $REMOTE_USER -b all -i $REMOTE_IP, -a'rm -f /var/lib/owca/lc-util.csv /var/lib/owca/workload-meta.json /var/lib/owca/workload-data.csv /var/lib/owca/threshold.json'
+
+
+
+3. Run application needed for `collect` mode.
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
3. Start application.
```

File: prm/demo_scenarios/README.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    ansible-playbook -l $REMOTE_IP -i demo_scenarios/complex_llc.0/inventory.yaml -i $cinventory $playbook \
+      --tags=twemcache_mutilate,redis_rpc_perf,cassandra_stress--cassandra
+
+    # or complex_mbw.0
+    ansible-playbook -l $REMOTE_IP -i demo_scenarios/complex_mbw.0/inventory.yaml -i $cinventory $playbook \
+      --tags=specjbb,tensorflow_benchmark_prediction,tensorflow_benchmark_training,cassandra_stress
+
+
+
+4. reconfigure (set action_delay to 20seconds) and restart OWCA+PRM to run in `colect` mode
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
4. Reconfigure OWCA (by setting `action_delay` to 20 seconds) and restart it to run in `collect` mode
```

File: prm/demo_scenarios/README.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+.. code-block:: sh
+
+    ansible -u $REMOTE_USER -b all -i $REMOTE_IP, -mreplace -a'path=/etc/owca/owca_config.yml after=detector regexp="detect" replace="collect"'
+    ansible -u $REMOTE_USER -b all -i $REMOTE_IP, -mreplace -a'path=/etc/owca/owca_config.yml regexp="action_delay: .*" replace="action_delay: 20."'
+    ansible -u $REMOTE_USER -b all -i $REMOTE_IP, -msystemd -a'name=owca state=restarted'
+
+
+
+5. create/kill the job observe anomaly disappers/appears
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]:
```suggestion
5. Keep creating and killing jobs to observe anomalies.
```

File: prm/demo_scenarios/README.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    ansible-playbook -l $REMOTE_IP -i $inventory $playbook --tags=cassandra_stress--stress -v
+    ansible-playbook -l $REMOTE_IP -i $inventory $playbook --tags=clean_jobs -ekill_job_name=cassandra_stress.default--cassandra_stress--9142.0 -v
+
+    # complex_mbw.0
+    ansible-playbook -l $REMOTE_IP -i $inventory $playbook --tags=tensorflow_benchmark_prediction -v
+    ansible-playbook -l $REMOTE_IP -i $inventory $playbook --tags=clean_jobs -ekill_job_name=tensorflow_benchmark_prediction.default--0.0 -v
+
+
+
+To check status:
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]: Could you elaborate a bit? Status of what you want to check?

File: prm/demo_scenarios/common/common.template.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Fill needed IPs
+
+application_hosts:
+  hosts:
+    100.64.176.13:
```
Comment (iwankgb) [inline prm/demo_scenarios/common/common.template.yaml]: See [RFC 5737](https://tools.ietf.org/html/rfc5737) for IP blocks to be used in the documentation.

File: prm/demo_scenarios/complex_llc.0/inventory.yaml [hunk 10]
```diff
@@ -0,0 +1,122 @@
+# @Scenario LLC contention:
+# 3 redis instances;
+# 3 twemcached instances (one big instance with big workload, other two relatively small);
+# start cassandra service but no workload to train the model;
+# start the cassandra stressor to test the model for cache contention.
+
+application_hosts:
+  hosts:
+    100.64.176.13:
```
Comment (iwankgb) [inline prm/demo_scenarios/complex_llc.0/inventory.yaml]: See [RFC 5737](https://tools.ietf.org/html/rfc5737) for IP blocks to be used in the documentation.

File: prm/demo_scenarios/complex_mbw.0/inventory.yaml [hunk 11]
```diff
@@ -0,0 +1,74 @@
+# @Scenario LLC contention:
+# 3 redis instances;
+# 3 twemcached instances (one big instance with big workload, other two relatively small);
+# start cassandra service but no workload to train the model;
+# start the cassandra stressor to test the model for cache contention.
+
+application_hosts:
+  hosts:
+    100.64.176.13:
```
Comment (iwankgb) [inline prm/demo_scenarios/complex_mbw.0/inventory.yaml]: See [RFC 5737](https://tools.ietf.org/html/rfc5737) for IP blocks to be used in the documentation.

File: prm/demo_scenarios/README.rst (hunk 8 above)
Comment (felidadae) [reply prm/demo_scenarios/README.rst]: fixed

File: prm/demo_scenarios/common/common.template.yaml (hunk 9 above)
Comment (felidadae) [reply prm/demo_scenarios/common/common.template.yaml]: fixed

File: prm/demo_scenarios/complex_llc.0/inventory.yaml (hunk 10 above)
Comment (felidadae) [reply prm/demo_scenarios/complex_llc.0/inventory.yaml]: fixed

File: prm/demo_scenarios/complex_mbw.0/inventory.yaml (hunk 11 above)
Comment (felidadae) [reply prm/demo_scenarios/complex_mbw.0/inventory.yaml]: fixed

File: prm/demo_scenarios/README.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
+Demo scenarios
+....................
+This document describes how to run demo scenarios on a Apache Mesos/Aurora cluster.
+
+Assumptions:
+
+- working directory: PRM project top level directory,
+- copied and filled the template file `demo_scenarios/common/common.template.yaml` to `demo_scenarios/common/common.yaml`,
```
Comment (iwankgb) [inline prm/demo_scenarios/README.rst]: Make `demo_scenarios/common/common.template.yaml` a link, please.

**Comment by MCamp859** [comment]
@felidadae  @iwankgb   I have an AR to edit/format the readmes in this project. I will also convert this RST file into MD format, OK?

**Comment by iwankgb** [comment]
@MCamp859 it is perfectly fine with me :)

**Pull request merged**